packages/data-store/src/xstate/XStateStore.ts

```diff
@@ -0,0 +1,86 @@
+import { OrPromise } from '@sphereon/ssi-types'
+import Debug from 'debug'
+import { DataSource, FindOptionsWhere, LessThan } from 'typeorm'
+
+import { StateEntity } from '../entities/xstate/StateEntity'
+import { DeleteExpiredStateArgs, DeleteStateArgs, GetStateArgs, GetStatesArgs, SaveStateArgs, State } from '../types'
+import { IAbstractXStateStore } from './IAbstractXStateStore'
+
+const debug = Debug('sphereon:ssi-sdk:xstate')
+
+export class XStateStore extends IAbstractXStateStore {
+  private readonly dbConnection: OrPromise<DataSource>
+
+  constructor(dbConnection: OrPromise<DataSource>) {
+    super()
+    this.dbConnection = dbConnection
+  }
+
+  async saveState(state: SaveStateArgs): Promise<State> {
+    const connection: DataSource = await this.dbConnection
+    debug(`Executing saveState with state: ${JSON.stringify(state)}`)
+    return connection.getRepository(StateEntity).save(state)
+  }
+
+  async getState(args: GetStateArgs): Promise<State> {
```

Comment:
I am not really sure about this method. It makes an assumption there will be only one state of a certain type. It does not take into account tenants at all and also does not look at expired states etc. IMO This method should return the most recent non-final Xstate instance of the type, but also look at the tenant and whether it is not final

Reply:
changed the way we're handling this. made this look into the `expiresAt` and `completedAt` fields and return the most recent one

Reply:
Much better. Although I am still a bit about the fence of returning only one entity.  I think we better return all of them, as I do believe that in the future we can have multiple instances of a flow running at the same time (for instance multiple people from the same tenant). I rather see us return all results. Any user of the method then can always get the first result or do some additional matching in the results

Reply:
Will change this to return multiple results so the user can choose

Reply:
It is now returning an array